light-prover/prover/circuit_builder.go

```diff
@@ -8,20 +8,31 @@ import (
 type CircuitType string
 
 const (
-	InputCompressedAccounts             = "input-compressed-accounts"
-	Combined                CircuitType = "combined"
-	Inclusion               CircuitType = "inclusion"
-	NonInclusion            CircuitType = "non-inclusion"
+	Combined     CircuitType = "combined"
+	Inclusion    CircuitType = "inclusion"
+	NonInclusion CircuitType = "non-inclusion"
+	BatchAppend  CircuitType = "append"
 )
 
-func SetupCircuit(circuit CircuitType, inclusionTreeDepth uint32, inclusionNumberOfCompressedAccounts uint32, nonInclusionTreeDepth uint32, nonInclusionNumberOfCompressedAccounts uint32) (*ProvingSystem, error) {
-	if circuit == Inclusion {
+func SetupCircuit(
+	circuit CircuitType,
+	inclusionTreeDepth uint32,
+	inclusionNumberOfCompressedAccounts uint32,
+	nonInclusionTreeDepth uint32,
+	nonInclusionNumberOfCompressedAccounts uint32,
+	batchAppendTreeDepth uint32,
+	batchAppendBatchSize uint32,
+) (*ProvingSystem, error) {
+	switch circuit {
+	case Inclusion:
 		return SetupInclusion(inclusionTreeDepth, inclusionNumberOfCompressedAccounts)
-	} else if circuit == NonInclusion {
+	case NonInclusion:
 		return SetupNonInclusion(nonInclusionTreeDepth, nonInclusionNumberOfCompressedAccounts)
-	} else if circuit == Combined {
+	case Combined:
 		return SetupCombined(inclusionTreeDepth, inclusionNumberOfCompressedAccounts, nonInclusionTreeDepth, nonInclusionNumberOfCompressedAccounts)
-	} else {
+	case BatchAppend:
+		return SetupBatchAppend(batchAppendTreeDepth, batchAppendBatchSize)
+	default:
```

Comment:
can we do refactors like this in a separate pr pls?

Reply:
https://github.com/Lightprotocol/light-protocol/pull/1284
^ let's merge that first, and then I rebase this over new main